test/functional/readconcern_tests.js

```diff
@@ -94,6 +52,107 @@ describe('ReadConcern', function() {
     }
   ];
 
+  describe('client-url specific ReadConcern', function() {
+    const urlTests = [
+      {
+        description: 'Should set local readConcern at collection level',
+        level: 'local',
+        commandName: 'find',
+        collectionReadConcern: true,
+        readConcern: { level: 'local' }
+      },
+      {
+        description: 'Should set majority readConcern at collection level',
+        level: 'majority',
+        commandName: 'find',
+        collectionReadConcern: true,
+        readConcern: { level: 'majority' }
+      },
+      {
+        description: 'Should set local readConcern using MongoClient',
+        level: 'local',
+        commandName: 'find',
+        urlReadConcernLevel: 'readConcernLevel=local',
+        readConcern: { level: 'local' }
+      },
+      {
+        description: 'Should set majority readConcern using MongoClient',
+        level: 'majority',
+        commandName: 'find',
+        urlReadConcernLevel: 'readConcernLevel=majority',
+        readConcern: { level: 'majority' }
+      },
+      {
+        description: 'Should set majority readConcern using MongoClient with options',
+        level: 'majority',
+        commandName: 'find',
+        urlOptions: 'majority',
+        readConcern: { level: 'majority' }
+      }
+    ];
+
+    urlTests.forEach(test => {
+      it(test.description, {
+        metadata: { requires: { topology: 'replicaset', mongodb: '>= 3.2' } },
+        test: function(done) {
+          var started = [];
+          var succeeded = [];
+          // Get a new instance
+          const configuration = this.configuration;
+          let options;
+          let url;
+
+          if (test.urlReadConcernLevel || test.urlOptions) {
```

Comment:
It seems like we're mainly trying to set the `url` based on `test.urlReadConcernLevel` here. If that's the case, can we just keep L107-L112? We could move the `options` definition directly to the new client construction.